src/Resolver/BounceTypeResolver.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AssoConnect\SmtpToolbox\Resolver;
+
+class BounceTypeResolver
+{
+    public const BOUNCE_REASON_UNQUALIFIED = 'BOUNCE_REASON_UNQUALIFIED';
+    public const BOUNCE_REASON_UNKNOWN = 'BOUNCE_REASON_UNKNOWN';
+    public const BOUNCE_REASON_DMARC_FAILURE = 'BOUNCE_REASON_DMARC_FAILURE';
+    public const BOUNCE_REASON_BLACKLISTED = 'BOUNCE_REASON_BLACKLISTED';
+    public const BOUNCE_REASON_GREYLISTING = 'BOUNCE_REASON_GREYLISTING';
+    public const BOUNCE_REASON_SPAMMY = 'BOUNCE_REASON_SPAMMY';
+    public const BOUNCE_REASON_DENIED = 'BOUNCE_REASON_DENIED';
+    public const BOUNCE_REASON_USER_ACTION_REQUIRED = 'BOUNCE_REASON_USER_ACTION_REQUIRED';
+    public const BOUNCE_REASON_INVALID = 'BOUNCE_REASON_INVALID';
+    public const BOUNCE_REASON_TEMPORARY = 'BOUNCE_REASON_TEMPORARY';
+    // phpcs:disable
+    public const BOUNCE_REASONS_MAPPING = [
```

Comment:
I hesitated, it complicates the actual function, where I'll have to do a foreach on each table. If a new table is added, it might be skipped in the checks.

Not so happy about it, but the alternatives did not seem a lot better

Reply:
Or a sub-array so you have two foreach?